unittests/failuretests.py

```diff
@@ -268,15 +268,13 @@ def test_subproject_variables(self):
         '''
         tdir = os.path.join(self.unit_test_dir, '20 subproj dep variables')
         stray_file = os.path.join(tdir, 'subprojects/subsubproject.wrap')
-        if os.path.exists(stray_file):
-            windows_proof_rm(stray_file)
+        self.addCleanup(windows_proof_rm, stray_file)
```

Comment:
This changes behaviour. The old code ensured that if the file was already there for whatever reason it would be deleted _before_ the test was run.

Reply:
Indeed, it's changing behavior. The current behavior is:
1. the test creates the file
2. the test runs, and makes no effort to delete the file
3. on re-run the test deletes the file
This is weird, and not something that we generally do.

This changes the behavior to match what basically all tests do, and what should happen:
1. the test creates the file
2. the test registers a callback to ensure that the file is cleaned up when the test exits, regardless of whether that is in error or in success
3. on re-run the test doesn't need to look for the file, because it doesn't exist.

We have *tons* of tests that make the assumption "files I create as part of my test process" do no pre-exist, and will die horribly, or worse, fail silently if they do.